src/components/datePicker/DatePickerReducers.ts

```diff
@@ -0,0 +1,112 @@
+import { IReduxAction } from '../../utils/ReduxUtils';
+import { IReduxActionsPayload } from '../../ReactVapor';
+import { DatePickerActions } from './DatePickerActions';
+import * as _ from 'underscore';
+import * as moment from 'moment';
+
+export interface IDatePickerState {
+  id: string;
+  calendarId: string;
+  color: string;
+  lowerLimit: Date;
+  upperLimit: Date;
+  isRange: boolean;
+  selected: string;
+  appliedLowerLimit: Date;
+  appliedUpperLimit: Date;
+}
+
+export const datePickerInitialState: IDatePickerState = {
+  id: undefined,
+  calendarId: undefined,
+  color: undefined,
+  isRange: false,
+  lowerLimit: moment().startOf('day').toDate(),
+  upperLimit: moment().endOf('day').toDate(),
+  selected: '',
+  appliedLowerLimit: moment().startOf('day').toDate(),
+  appliedUpperLimit: moment().endOf('day').toDate()
+};
+export const datePickersInitialState: IDatePickerState[] = [];
+
+export const datePickerReducer = (state: IDatePickerState = datePickerInitialState,
+  action: IReduxAction<IReduxActionsPayload>): IDatePickerState => {
```

Comment:
I thinks you could rework the IReduxActionsPayload interface and replace it with a union of all interfaces possible to use here, like I did in sources security identities. It would be better.